src/vcpkg/commands.install.cpp

```diff
@@ -46,57 +46,59 @@ namespace vcpkg
 
     const Path& InstallDir::listfile() const { return this->m_listfile; }
 
-    void install_package_and_write_listfile(const Filesystem& fs,
-                                            const Path& source_dir,
-                                            const InstallDir& destination_dir)
+    static void install_package_and_write_listfile(const Filesystem& fs,
+                                                   const Path& source_dir,
+                                                   const InstallDir& destination_dir)
     {
         Checks::check_exit(VCPKG_LINE_INFO,
                            fs.exists(source_dir, IgnoreErrors{}),
                            Strings::concat("Source directory ", source_dir, "does not exist"));
-        auto files = fs.get_files_recursive(source_dir, VCPKG_LINE_INFO);
-        Util::erase_remove_if(files, [](Path& path) { return path.filename() == FileDotDsStore; });
-        install_files_and_write_listfile(fs, source_dir, files, destination_dir);
+        auto source_suffixes = fs.get_files_recursive_lexically_proximate(source_dir, VCPKG_LINE_INFO);
+        Util::erase_remove_if(source_suffixes, [](Path& path) {
+            path.make_forward_slashes();
```

Comment:
Why is this better than `make_generic()`?

Reply:
`make_generic` does a whole bunch of processing to handle the `\\server\share` case we don't care about because we know these will be relative paths:

https://github.com/microsoft/vcpkg-tool/blob/61d0793bafdb3040caca7061fee241264813a067/src/vcpkg/base/files.cpp#L1164-L1202